### PR TITLE
Streaming Kaldi feature extractors

### PR DESCRIPTION
**Pull request by pzelasko**
This updates the Kaldi-like feature extractor layers with a new function `online_inference`. It accepts an extra argument which is the remainder of the previous waveform (or `None`) and it returns the remainder from the currently processed audio chunk. It works basically the same like Kaldi's online feature extraction, except it has a stateless API which plays better with TorchScript and other runtimes. The callers should manage the state themselves.

### CODE REVIEW

**Comment by pzelasko** [comment]
I was able to replicate the WER of a pretrained ASR model after these modifications.

File: lhotse/features/kaldi/layers.py [hunk 1]
```diff
@@ -13,10 +13,17 @@
 
 This implementation works well with autograd and batching, and can be used neural network
 layers.
+
+Update January 2022:
+These modules now expose a new API function called "online_inference" that
+may be used to compute the features when the audio is streaming.
+The implementation is stateless, and passes the waveform remainders
+back to the user to feed them to the modules once new data becomes available.
+The implementation is compatible with JIT scripting via TorchScript.
```
Comment (csukuangfj) [inline lhotse/features/kaldi/layers.py:22]: Are there tests to ensure this?

Comment (pzelasko) [reply lhotse/features/kaldi/layers.py:22]: Yeah, this PR #521 added torchscript compatibility and tests for it, and in this PR the new methods are annotated with `torch.jit.export`, so they're automatically tested by the previously existing tests (I also manually torchscripted it and ran some forward passes to make sure).

**Comment by danpovey** [comment]
A great feature!

**Comment by pzelasko** [comment]
I refined it to give the same results as the torchaudio extractors with the default settings used in Lhotse. Merging.

**Pull request merged**